File: backend/src/modules/configuracion/configuracion-publica.controller.ts
import { Controller, Get, HttpCode, HttpStatus } from '@nestjs/common';
import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
import { ConfiguracionTiendaService } from './configuracion-tienda.service';

@ApiTags('configuracion-tienda')
@Controller('configuracion-tienda')
export class ConfiguracionPublicaController {
  constructor(private readonly configuracionService: ConfiguracionTiendaService) {}

  @Get('publica')
  @HttpCode(HttpStatus.OK)
  @ApiOperation({ summary: 'Obtener datos publicos de la tienda' })
  @ApiResponse({ status: 200, description: 'Datos publicos de la tienda' })
  async obtenerPublica() {
    const config = await this.configuracionService.obtener();
    return {
      nombreTienda: config.nombreTienda,
      whatsappContacto: config.whatsappContacto,
    };
  }
}

File: backend/src/main/java/com/sistemapedidos/modules/configuracion/ConfiguracionPublicaController.java
package com.sistemapedidos.modules.configuracion;

import com.sistemapedidos.modules.configuracion.dto.ConfiguracionTiendaResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/configuracion-tienda")
@RequiredArgsConstructor
public class ConfiguracionPublicaController {

    private final ConfiguracionTiendaService configuracionService;

    @GetMapping("/publica")
    public ConfiguracionTiendaResponse.ConfiguracionPublica obtenerPublica() {
        ConfiguracionTiendaResponse config = configuracionService.obtener();
        return new ConfiguracionTiendaResponse.ConfiguracionPublica(
                config.nombreTienda(),
                config.whatsappContacto());
    }
}
